File: world_simulation/grass.py
from sun import Sun

class Grass:
    def __init__(self, quantity: int = 100) -> None:
        self._quantity = quantity
    
    def give_grass(self, quantity: int = 1) -> int:
        if self._quantity >= quantity:
            self._quantity -= quantity
            return quantity
        else:
            raise ValueError("Can't eat more grass than exists.")

    def grow(self, sun: "Sun") -> None:
        if sun.is_shining():
            self._quantity += 2

    def get_quantity(self) -> int:
        return self._quantity

    def print_status(self) -> None:
        print(f"Grass quantity is: {self.get_quantity()}")

File: world_simulation_c/grass.c
#include "sun.h"
#include "grass.h"
#include <stdio.h>
#include <stdlib.h>

typedef struct Grass {
    int quantity;
    int grow_rate;
} Grass;

Grass * grass_init(){
    return (Grass*)malloc(sizeof(Grass));
}

void grass_construct(Grass * g, int quantity, int grow_rate){
    g->quantity = quantity;
    g->grow_rate = grow_rate;
}

void grass_destruct(Grass * g){
    free(g);
}

void grow(Grass * g, struct Sun * s){
    if(is_shining(s)){
        g->quantity += g->grow_rate;
    }
}

int give_grass(Grass * g, int quantity){
    g->quantity -= quantity;
    return quantity;
}

void grass_print_status(Grass * g){
    printf("Grass quantity is %d.\n", g->quantity);
}